homeassistant/components/automation/__init__.py

```diff
@@ -272,15 +273,29 @@ async def trigger_service_handler(
         "async_turn_off",
     )
 
-    async def reload_service_handler(service_call: ServiceCall) -> None:
+    async def reload_all_service_handler(service_call: ServiceCall) -> None:
         """Remove all automations and load new ones from config."""
         if (conf := await component.async_prepare_reload()) is None:
             return
         async_get_blueprints(hass).async_reset_cache()
-        await _async_process_config(hass, conf, component)
+        await _async_process_config(hass, conf, component, None)
         hass.bus.async_fire(EVENT_AUTOMATION_RELOADED, context=service_call.context)
 
-    reload_helper = ReloadServiceHelper(reload_service_handler)
+    async def reload_single_service_handler(service_call: ServiceCall) -> None:
+        """Remove all automations and load new ones from config."""
+        automation_id = service_call.data[CONF_ID]
+        automation = component.get_entity(automation_id)
+        if automation:
+            await automation.async_remove()
+
+        if (conf := await component.async_prepare_reload(skip_reset=True)) is None:
+            return
+        async_get_blueprints(hass).async_reset_cache()
```

Comment:
Is this unnecessary?

Reply:
I don't think so. The blueprint could have changed as well, influencing the automation possibly.